src/node/nodes.h

```diff
@@ -48,15 +47,43 @@ namespace ccf
         trusted */
     std::optional<kv::Version> ledger_secret_seqno = std::nullopt;
 
-    /** Code identity for the node **/
-    std::optional<std::string> code_digest;
+    /// Code identity for the node
+    std::optional<std::string> code_digest = std::nullopt;
+
+    /**
+     *  Fields below are added in 2.x
+     */
+
+    /// Node certificate signing request
+    std::optional<crypto::Pem> certificate_signing_request = std::nullopt;
+
+    /// Public key
+    std::optional<crypto::Pem> public_key = std::nullopt;
+
+    /**
+     * Fields below are deprecated
+     */
+
+    /** Deprecated as of 2.x.
+     * Node certificate. Only set for 1.x releases. Further releases record
+     * node identity in `public_key` field. Service-endorsed certificate is
+     * recorded in "public:ccf.nodes.endorsed_certificates" table */
+    std::optional<crypto::Pem> cert = std::nullopt;
```

Comment:
Moving a previously required field to optional here. This means that it can be parsed from the historical ledger, but isn't serialised in the ledger for new node entries that don't use this field.